components/BasicItemSlot.jsx

```diff
@@ -0,0 +1,90 @@
+import styled from "styled-components";
+import basicItemsList from "../data/basicitems-data.json";
+import slugify from "slugify";
+
+export const BasicItemSlot = ({ register, index, watch, basicItemRemove }) => {
```

Comment:
I think it would make sense to create one generic slot component that abstracts items, spells, etc.. Let's talk if you'd like hints on how to do that. It's  not super necessary for this PR though.